netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -351,22 +351,20 @@ public void run() {
     }
 
 
-    if (transportTracer != null) {
-      assert encoder().connection().equals(decoder().connection());
-      final Http2Connection connection = encoder().connection();
-      transportTracer.setFlowControlWindowReader(new TransportTracer.FlowControlReader() {
-        private final Http2FlowController local = connection.local().flowController();
-        private final Http2FlowController remote = connection.remote().flowController();
+    assert encoder().connection().equals(decoder().connection());
+    final Http2Connection connection = encoder().connection();
```

Comment:
this could be non final if you extracted local and remove up a scope level.

Reply:
done